app_server.py

```diff
@@ -239,218 +239,108 @@ def heartbeat(self):
-                identifier,
-                ", ".join(names)
-            )
-            unresolved_identifier, is_new = UnresolvedIdentifier.register(
-                self._db, identifier)
-            if is_new:
-                # We just found out about this identifier, or rather,
-                # we just found out that someone expects it to be associated
-                # with a LicensePool.
-                return (201, self.IDENTIFIER_REGISTERED)
-            else:
-                # There is a pending attempt to resolve this identifier.
-                message = (unresolved_identifier.exception 
-                           or self.WORKING_TO_RESOLVE_IDENTIFIER)
-                return (unresolved_identifier.status, message)
-        else:
-            # All metadata lookups have completed. Create that OPDS
-            # entry!
-            entry = identifier.opds_entry()
-
-        if entry is None:
-            # This app can't do lookups on an identifier of this
-            # type, so the best thing to do is to treat this
-            # identifier as a 404 error.
-            return (404, self.UNRECOGNIZED_IDENTIFIER)
-
-        # We made it!
-        return entry
-
-    def work_lookup(self, annotator, route_name='lookup',
-                    require_active_licensepool=True, collection=None):
+    def work_lookup(self, annotator, route_name='lookup', collection=None,
+                    urns=[]):
         """Generate an OPDS feed describing works identified by identifier."""
         urns = flask.request.args.getlist('urn')
 
         this_url = cdn_url_for(route_name, _external=True, urn=urns)
         for urn in urns:
-            self.process_urn(urn, collection=collection)
-
-        # The commit is necessary because we may have registered new
-        # Identifier or UnresolvedIdentifier objects.
-        self._db.commit()
+            self.process_urn(urn, collection)
+        self.post_lookup_hook()
 
         opds_feed = LookupAcquisitionFeed(
             self._db, "Lookup results", this_url, self.works, annotator,
             messages_by_urn=self.messages_by_urn, 
             precomposed_entries=self.precomposed_entries,
-            require_active_licensepool=require_active_licensepool
         )
         return feed_response(opds_feed)
 
-    def permalink(self, urn, annotator):
-        """Generate an OPDS feed for looking up a single work by identifier."""
-        this_url = cdn_url_for('work', _external=True, urn=urn)
+    def permalink(self, urn, annotator, route_name='work'):
+        """Look up a single identifier and generate an OPDS feed."""
+        this_url = cdn_url_for(route_name, _external=True, urn=urn)
         self.process_urn(urn)
+        self.post_lookup_hook()
 
-        # The commit is necessary because we may have registered new
-        # Identifier or UnresolvedIdentifier objects.
-        self._db.commit()
-
+        # A LookupAcquisitionFeed's .works is a list of (identifier,
```

Comment:
Based on this mixup, I looked around, and `URNLookupController#permalink` isn't being used anywhere. It's fine to leave it, but it's prime for the chopping block.